### ISSUE TITLE
[TODO] Add include collection header-file with namespace shortcuts.

### ISSUE THREAD

**Issue by lganzzzo** [oatpp/oatpp#355]
Create `.hpp` file with most used includes + shortcut typedefs.

Example:

let's use `oatpp::http::IncomingRequest` instead of `oatpp::web::protocol::http::incoming::Request`

```cpp
#include "oatpp/web/protocol/http/incoming/Request"
...

namespace oatpp { namespace http {
  typedef oatpp::web::protocol::http::incoming::Request IncomingRequest;
  typedef oatpp::web::protocol::http::incoming::Response IncomingResponse;
  typedef oatpp::web::protocol::http::outgoing::Request OutgoingRequest;
  typedef oatpp::web::protocol::http::outgoing::Response OutgoingResponse;
}}

...
```

@bamkrs

**Comment by bamkrs** [comment]
Good idea! I like the idea to keep the essential parts as namespace portions (`oatpp::http`) and especially like the new naming scheme for the "sub-types" (`incoming::Request` to `IncomingRequest`). This feels way more natural! 
We should especially keep an eye on the "frequently used" stuff like headers, json-mapper, and everything APIController-related. Maybe shorting the essentials all together:

`oatpp::web::server::api::ApiController` -> `oatpp::ApiController`
`oatpp::web::client::ApiClient` -> `oatpp::ApiClient`

---
Maybe dont squash them into one single header but use a collection of headers:

* `ApiControllerShortcuts.hpp`
* `ServerConnectionShortcuts.hpp` (Connectionprovider etc...)

Thus we could create a bunch of shortcuts and only include those that are needed. We would circumvent problems with auto-completion because of `oatpp::` getting way to large.
